salt/modules/selinux.py

```diff
@@ -469,11 +469,17 @@ def fcontext_get_policy(name, filetype=None, sel_type=None, sel_user=None, sel_l
     current_entry_text = __salt__['cmd.shell'](cmd, ignore_retcode=True)
     if current_entry_text == '':
         return None
+
+    current_entry_list = current_entry_text.split(' ')
+    filespec = current_entry_list.pop(0)
+    context_string = current_entry_list.pop()
+    filetype = ' '.join(current_entry_list).strip()
```

Comment:
I think there is a better way to do this.

```
filespec, ftype1, ftype2, context_string  = re.split(re_spacer, current_entry_text)
ret = {
    'filespec': filespec,
    'filetype': ' '.join((ftype1, ftype2)),
}
ret.update(_context_string_to_dict(context_string))
```

Reply:
Unfortunately this won't work, as the filetype _may_ contain a space, but doesn't always.

Reply:
An alternative though (untested pseudocode):

```
parts = re.match('({filespec}) +([a-z ]+) +(.*)'.format({'filespec': re.escape(name)}))
ret = {
    'filespec': parts.group(1),
    'filetype': parts.group(2),
}
ret.update(_context_string_to_dict(parts.group(3)))
```

So parsing using a regular expression groups.

Reply:
This looks better to me :+1: